src/mmda/types/document.py

```diff
@@ -160,37 +159,61 @@ def _annotate_box_group(
 
         all_page_tokens = dict()
         derived_span_groups = []
+        token_box_in_box_group = None
 
         for box_id, box_group in enumerate(box_groups):
 
-            all_token_spans_with_box_group = []
+            all_tokens_overlapping_box_group = []
 
             for box in box_group.boxes:
 
                 # Caching the page tokens to avoid duplicated search
                 if box.page not in all_page_tokens:
-                    cur_page_tokens = all_page_tokens[box.page] = list(
-                        itertools.chain.from_iterable(
-                            span_group.spans
-                            for span_group in self.pages[box.page].tokens
+                    cur_page_tokens = all_page_tokens[box.page] = self.pages[
+                        box.page
+                    ].tokens
+                    if token_box_in_box_group is None:
+                        # Determine whether box is stored on token SpanGroup span.box or in the box_group
```

Comment:
can we add a `logging.warning` if they go down the route of `token SpanGroup span.box` that we will deprecate support for this and that they should make sure `Document` is annotated with `tokens` with `Box Groups`

Reply:
I think we should do this but wait till we get closer to the mmda refactor work being taken on. This will produce A LOT of logs and might blow up datadog if we let it sit for months on end